src/backend/WorkflowManagement/WorkflowManager.cs

```diff
@@ -76,10 +77,10 @@ public static WorkflowQueueMessage GetFirstStage(string submissionId)
         /// Executes the action for the current stage in the workflow. After the action completes,
         /// returns a message for the next stage in the queue. 
         /// </summary>
-        public async Task<WorkflowQueueMessage> ExecuteActionsToNextStage(WorkflowQueueMessage currentMsg)
+        public async Task<WorkflowQueueMessage> ExecuteActionsToNextStage(WorkflowQueueMessage currentMsg, CancellationToken cancelToken)
         {
             //Execute the action
-            WorkflowStage nextStage = await actions[(int)currentMsg.Stage].ExecuteAsync(currentMsg.SubmissionId);
+            WorkflowStage nextStage = await actions[(int)currentMsg.Stage].ExecuteAsync(currentMsg.SubmissionId, cancelToken);
```

Comment:
Do you need any bounds checking here? Such as `actions.TryGetValue` or something?

Reply:
Yes, I've been meaning to improve that...thanks for bringing it up.